### ISSUE TITLE
[flutter_tools] ProcessException: Failed to find "C:\flutter\bin\cache\dart-sdk\bin\dartaotruntime.exe" in the search path

### ISSUE THREAD

**Issue by christopherfujino** [flutter/flutter#143391]
Top crasher on `beta/3.19.0-0.4.pre`

Command: `flutter --no-color run --machine --track-widget-creation --device-id=0c463b4b0602 --start-paused --dart-define=flutter.inspector.structuredErrors=true lib\main.dart`

```
ProcessPackageException: ProcessException: Failed to find "C:\flutter\bin\cache\dart-sdk\bin\dartaotruntime.exe" in the search path.  Command: C:\flutter\bin\cache\dart-sdk\bin\dartaotruntime.exe
at LocalProcessManager.start(local_process_manager.dart:56)
at ErrorHandlingProcessManager.start.<anonymous closure>(error_handling_io.dart:686)
at _run(error_handling_io.dart:556)
at ErrorHandlingProcessManager.start(error_handling_io.dart:685)
at DefaultResidentCompiler._compile(compile.dart:874)
at DefaultResidentCompiler._recompile(compile.dart:734)
at _RecompileRequest._run(compile.dart:420)
at _CompilationRequest.run(compile.dart:394)
at DefaultResidentCompiler._handleCompilationRequest(compile.dart:780)
at _rootRunUnary(zone.dart:1407)
at _CustomZone.runUnary(zone.dart:1308)
at _CustomZone.runUnaryGuarded(zone.dart:1217)
at _BufferingStreamSubscription._sendData(stream_impl.dart:339)
at _DelayedData.perform(stream_impl.dart:515)
at _PendingEvents.handleNext(stream_impl.dart:620)
at _PendingEvents.schedule.<anonymous closure>(stream_impl.dart:591)
at _rootRun(zone.dart:1391)
at _CustomZone.run(zone.dart:1301)
at _CustomZone.runGuarded(zone.dart:1209)
at _CustomZone.bindCallbackGuarded.<anonymous closure>(zone.dart:1249)
at _rootRun(zone.dart:1399)
at _CustomZone.run(zone.dart:1301)
at _CustomZone.runGuarded(zone.dart:1209)
at _CustomZone.bindCallbackGuarded.<anonymous closure>(zone.dart:1249)
at _microtaskLoop(schedule_microtask.dart:40)
at _startMicrotaskLoop(schedule_microtask.dart:49)
at _runPendingImmediateCallback(isolate_patch.dart:118)
at _RawReceivePort._handleMessage(isolate_patch.dart:185)
```

**Comment by christopherfujino** [comment]
The reports of this I'm seeing are all on Windows. However, when I open the pre-built windows zip for this release from https://storage.googleapis.com/flutter_infra_release/releases/beta/windows/flutter_windows_3.19.0-0.4.pre-beta.zip, I do indeed see this binary file present.

**Comment by AndreLuizNogueira** [comment]
I'm having the same problem. my antivirus is deleting it... only on the beta channel. I use Kaspersky 21.3.10.391 on Windows 11 pro 22H2.
for me it started on 14/02/2024 at around 03:30 GMT -3
I checked on virustotal and got 4 results
[https://www.virustotal.com/gui/file/40d6eb32a1e384a9e05c3cbf3d0b843d7b1aa70e1e474e4dcfd660e810d38af8?nocache=1](url)

**Comment by christopherfujino** [comment]
I had concluded that this was windows defender, but perhaps it's coming from alternative virus scans (I can't repro it with defender).

@AndreLuizNogueira will Kaspersky let you allowlist a binary? Also, this binary exists on master as it does in beta, I suspect maybe the difference is the way you downloaded it? If so, switching back to beta and trying it again may work.

**Comment by AndreLuizNogueira** [comment]
> I had concluded that this was windows defender, but perhaps it's coming from alternative virus scans (I can't repro it with defender).
> 
> @AndreLuizNogueira will Kaspersky let you allowlist a binary? Also, this binary exists on master as it does in beta, I suspect maybe the difference is the way you downloaded it? If so, switching back to beta and trying it again may work.

@christopherfujino 
I tried the exclusion list, but Kaspersky still deleted it. I've been back and forth between stable, beta, and main... every time I go to beta, it's quarantined. I tried to get it from a coworker who doesn't have antivirus and I can't even download it on Chrome

**Comment by ashokavarthanan** [comment]
I'm also having the same problem. my AVG antivirus is deleted some files.
Now solved by this command.
```
flutter downgrade 
flutter upgrade --force
```

**Comment by pierrecorsini** [comment]
The same is happening to us right now (W3 2025), I'm not sure if this is due to a update of W11 or Acronis Cyber Protect, but this was not happening W1 of 2025.

<details>
<summary>Antivirus detection </summary>

```
Security analysis
Verdict: Malicious threat
Severity: HIGH
File found on
2 workloads
Technique[User Execution](https://attack.mitre.org/techniques/T1204/)
Tactic: Execution
Reason of detection: An adversary may rely upon specific actions by a user in order to gain execution. Users may be subjected to social engineering to get them to execute malicious code by, for example, opening a malicious document file or link. These user actions will typically be observed as follow-on behavior from forms of Phishing.
Detections
Malware detected
Detection date: Jan 17, 2025 11:05:29:545354
Threat name
View all threat names
ML:Generic.MaliciousExe
```

</details>

**Comment by oremlawi** [comment]
Hey guys, for this error, I've solved it in this way:
I have '**fvm**' to manage multiple versions for **Flutter** (in my case was **3.27.4**), so when I download manually the **Flutter** sdk from their [official page](https://docs.flutter.dev/release/archive), the '**dartaotruntime**' file is really missing from **dart-sdk/bin**, so I did was the following:

1. Remove manually the wrong Flutter sdk
2. Run this command: `fvm remove 3.27.4`  (replace '3.27.4' by your version)
3. Run this command: `fvm install 3.27.4`
4. Inside your project run `fvm use 3.27.4` and then run this `fvm flutter doctor -v` and it will download all missing files for you (including **dartaotruntime** in its correct location), and this is the trick.

**Comment by bkonyi** [comment]
> Hey guys, for this error, I've solved it in this way: I have '**fvm**' to manage multiple versions for **Flutter** (in my case was **3.27.4**), so when I download manually the **Flutter** sdk from their [official page](https://docs.flutter.dev/release/archive), the '**dartaotruntime**' file is really missing from **dart-sdk/bin**, so I did was the following:
> 
> 1. Remove manually the wrong Flutter sdk
> 2. Run this command: `fvm remove 3.27.4`  (replace '3.27.4' by your version)
> 3. Run this command: `fvm install 3.27.4`
> 4. Inside your project run `fvm use 3.27.4` and then run this `fvm flutter doctor -v` and it will download all missing files for you (including **dartaotruntime** in its correct location), and this is the trick.

@oremlawi I downloaded the Windows `3.27.4` stable release from the official page, `dartaotruntime.exe` is present at `flutter/bin/cache/dart-sdk/bin/dartaotruntime.exe`. You're saying that wasn't there for you?

**Comment by oremlawi** [comment]
Yes indeed, maybe because I'm on a MacBook M1, and even the latest version (so far) 3.29.0 arm64, the same file was missing!
